### ISSUE TITLE
better zoom lock

### ISSUE THREAD

**Issue by aeosynth** [Narrat/Pim#13]
i want zoom lock to make all images fit/full screen, instead of locking the zoom level

**Comment by Narrat** [comment]
So, pictures which are way smaller than the window size of pim should be stretched and reach that goal too?

**Comment by aeosynth** [comment]
if zoom is locked to fit screen, yes. maybe have an option to choose between lock percentage or lock mode

**Comment by Narrat** [comment]
or as a separate function with extra shortcut.
Was my thought as I like the current behaviour. Although not really the best thing. Likely it could cause some confusion.

Your suggestion could be a viable option

**Comment by Narrat** [comment]
Currently working on this, as it is already mostly implemented.
But working on it I noticed I need to redo the zoom handling stuff as it is currently easy to run into errors (as it happened while implementing it).

tl;dr: Kinda there, but needs some rework to implement it in a kina nicer way

**Comment by aeosynth** [comment]
sure no rush. you can tag a new release once a few of these issues are resolved
